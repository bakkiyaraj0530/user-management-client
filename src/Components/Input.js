import React from 'react';

function Input(props) {
  return (
      <input
        className="form-control form-control-lg"
        name={props.name}
        type={props.inputType}
        value={props.email}
        onChange={props.onChange}
        placeholder={props.placeholder}
      />
  );
}

export default Input;
