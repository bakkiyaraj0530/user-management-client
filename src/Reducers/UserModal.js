import { HANDLE_MODAL } from './../Constants';

export default (state = null, action) => {
    switch (action.type) {
        case HANDLE_MODAL:
            return action.payload;
        default:
            return state;
    }
};