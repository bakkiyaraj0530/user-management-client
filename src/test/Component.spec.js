import React from 'react';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import jsdom from 'jsdom';
import 'jsdom-global/register';

const mockStore = configureMockStore();
const store = mockStore({});


import { shallow, mount } from 'enzyme';
import GroupsList from './../Components/GroupsList';
import Header from './../Components/Header';
import Input from './../Components/Input';
import UserModal from './../Components/UserModal';
import UsersList from './../Components/UsersList';

// const doc = jsdom.jsdom('<!doctype html><html><body></body></html>');
// global.document = doc;
// global.window = doc.defaultView;

// import { findByTestAtrr, testStore } from './../Utils';

// const setUp = (initialState = {}) => {
//     const store = testStore(initialState);
//     const wrapper = shallow(<GroupsList store={store} />).childAt(0).dive();
//     return wrapper;
// };

// import TestPage from "../TestPage";


describe("Group listing page Component", () => {
    let component;
    let component2;
    let component3;
    let component4;

    let propsCall;
    const groups = {
        created: "2019-08-20T12:54:11.367Z",
        title: "ReactDev2",
        v: 0,
        id: "5d5bed73aaebcf01d494798c"
    };
    beforeEach(() => {

        propsCall = { handleLeft: jest.fn() }
        component = shallow(
            <Provider store={store}>
                <GroupsList groups={groups} />
            </Provider>)
        component2 = shallow(
            <Provider store={store}>
                <Header propsCall={propsCall} />
            </Provider>)
    });

    it('renders correctly', () => {
        const wrapper = shallow(<Input />);
        expect(wrapper).toMatchSnapshot();
    });
    describe('when Header Icon clicks ', () => {

        it('matches Header snapshot', () => {
            const onIconClickMock = jest.fn();

            const wrapper = shallow(
                <Provider store={store}>
                    <Header handleLeft={onIconClickMock} />
                </Provider>
            );
            expect(component.exists()).toBe(true);

            // wrapper.find('fa fa-bars').simulate('click');
            // expect(onIconClickMock).toHaveBeenCalledTimes(1); 
            // expect(onIconClickMock).toHaveBeenCalledWith(true);
        })
    });
    it('handle inner Method', () => {
        expect(component2.find('header').length).toBe(0);
    });
    it('matches Gouplist snapshot', () => {
        expect(component).toMatchSnapshot()
    })

    it("should render without throwing an error", () => {
        expect(component.exists()).toBe(true);
    });

    it('changes state when typing', () => {
        //     // component.state
        component = shallow(
            <Provider store={store}>
                <GroupsList groups={groups} />
            </Provider>)
        // component.instance().handleUpdate = jest.fn();
        expect(component.find('input').length).toBe(0);

        // expect(component.find('input').length).toBe(1)
        // expect(component.find('button').length).toBe(1)
        // expect(component.find('button').first().hasClass('btn'))
        // expect(component.find('button').first().hasClass('btn-success'))
        //     // expect(component.state('text')).toBe('Hello');

        //     expect(component.state('text')).toBe('Hello');
    })
    // })

});

