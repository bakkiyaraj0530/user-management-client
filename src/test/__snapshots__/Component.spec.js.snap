// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group listing page Component matches Gouplist snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
>
    <Connect(GroupsList)
        groups={
            Object {
                "created": "2019-08-20T12:54:11.367Z",
                "id": "5d5bed73aaebcf01d494798c",
                "title": "ReactDev2",
                "v": 0,
              }
        }
    />
</Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "groups": Object {
        "created": "2019-08-20T12:54:11.367Z",
        "id": "5d5bed73aaebcf01d494798c",
        "title": "ReactDev2",
        "v": 0,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "groups": Object {
          "created": "2019-08-20T12:54:11.367Z",
          "id": "5d5bed73aaebcf01d494798c",
          "title": "ReactDev2",
          "v": 0,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Group listing page Component matches Header snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
>
    <Connect(Header)
        propsCall={
            Object {
                "handleLeft": [Function],
              }
        }
    />
</Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "propsCall": Object {
        "handleLeft": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "propsCall": Object {
          "handleLeft": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Group listing page Component matches its snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
>
    <Connect(GroupsList)
        groups={
            Object {
                "created": "2019-08-20T12:54:11.367Z",
                "id": "5d5bed73aaebcf01d494798c",
                "title": "ReactDev2",
                "v": 0,
              }
        }
    />
</Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "groups": Object {
        "created": "2019-08-20T12:54:11.367Z",
        "id": "5d5bed73aaebcf01d494798c",
        "title": "ReactDev2",
        "v": 0,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "groups": Object {
          "created": "2019-08-20T12:54:11.367Z",
          "id": "5d5bed73aaebcf01d494798c",
          "title": "ReactDev2",
          "v": 0,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Group listing page Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "form-control form-control-lg",
      "name": undefined,
      "onChange": undefined,
      "placeholder": undefined,
      "type": undefined,
      "value": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "form-control form-control-lg",
        "name": undefined,
        "onChange": undefined,
        "placeholder": undefined,
        "type": undefined,
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Group listing page Component should render without throwing an error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
>
    <Connect(GroupsList)
        groups={
            Object {
                "created": "2019-08-20T12:54:11.367Z",
                "id": "5d5bed73aaebcf01d494798c",
                "title": "ReactDev2",
                "v": 0,
              }
        }
    />
</Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "groups": Object {
        "created": "2019-08-20T12:54:11.367Z",
        "id": "5d5bed73aaebcf01d494798c",
        "title": "ReactDev2",
        "v": 0,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "groups": Object {
          "created": "2019-08-20T12:54:11.367Z",
          "id": "5d5bed73aaebcf01d494798c",
          "title": "ReactDev2",
          "v": 0,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Testpage Component should render without throwing an error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
>
    <Connect(GroupsList)
        groups={
            Object {
                "created": "2019-08-20T12:54:11.367Z",
                "id": "5d5bed73aaebcf01d494798c",
                "title": "ReactDev2",
                "v": 0,
              }
        }
    />
</Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "groups": Object {
        "created": "2019-08-20T12:54:11.367Z",
        "id": "5d5bed73aaebcf01d494798c",
        "title": "ReactDev2",
        "v": 0,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "groups": Object {
          "created": "2019-08-20T12:54:11.367Z",
          "id": "5d5bed73aaebcf01d494798c",
          "title": "ReactDev2",
          "v": 0,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
